src/components/ContactsAdd.js

```diff
@@ -1,39 +1,145 @@
-import { useState } from "react"
+import { useState } from "react";
 import { useNavigate } from "react-router-dom";
 
-function ContactsAdd(props) {
-
+function ContactsAdd({ setContacts }) {
   // setContacts and contacts must be passed as props
   // to this component so new contacts can be added to the
+  const [newContact, setNewContact] = useState();
   // state
-  const { setContacts, contacts } = props
+  // const { setContacts, contacts } = props
 
   //TODO: Implement controlled form
   //send POST to json server on form submit
+  // console.log(setContacts);
+
+  const navigate = useNavigate();
+
+  const handleChange = (e) => {
+    const { name, value, type } = e.target;
+    switch (name) {
+      case "firstName":
+        return setNewContact({ ...newContact, firstName: value });
+        break;
+      case "lastName":
+        return setNewContact({ ...newContact, lastName: value });
+        break;
+      case "street":
+        return setNewContact({ ...newContact, street: value });
+        break;
+      case "city":
+        return setNewContact({ ...newContact, city: value });
+        break;
+      case "email":
+        return setNewContact({ ...newContact, email: value });
+        break;
+      case "linkedIn":
+        return setNewContact({ ...newContact, linkedIn: value });
+        break;
+      case "twitter":
+        return setNewContact({ ...newContact, twitter: value });
+        break;
+      case "type":
+        return setNewContact({ ...newContact, type: value });
+        break;
+    }
```

Comment:
I've done it the exact same way, but @Dan2024 has a great, cleaner code for this! FYI

Reply:
@sigristarisa i believe what i used is called computed property syntax, but yeah cases like this it can be really useful https://eloquentcode.com/computed-property-names-in-javascript